libsoftwarecontainer/unit-test/iptableentry_unittest.cpp

```diff
@@ -0,0 +1,100 @@
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#include "gateway/iptableentry.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+class MockIPTableEntry :
+    public IPTableEntry
+{
+public:
+    MockIPTableEntry():IPTableEntry()
+    {
+        m_type = "INPUT";
+        m_defaultTarget = Target::ACCEPT;
+    }
+};
+
+class IPTableEntryTest : public ::testing::Test
+{
+protected:
+    ::testing::NiceMock<MockIPTableEntry> ipTable;
+};
+
+//test policies
+TEST_F(IPTableEntryTest, TestIPTableEntryPolicies) {
+    IPTableEntry::Rule r;
+    r.host = "127.0.0.1/16";
+    r.ports.any = true;
+    r.ports.multiport = true;
+    r.ports.ports = "80,8080";
+    r.target = IPTableEntry::Target::ACCEPT;
+    ASSERT_EQ("iptables -P INPUT ACCEPT", ipTable.interpretPolicy());
+
+    ipTable.m_defaultTarget = IPTableEntry::Target::DROP;
+    ASSERT_EQ("iptables -P INPUT DROP", ipTable.interpretPolicy());
+
+    //Test wrong default Policy
+    ipTable.m_defaultTarget = IPTableEntry::Target::REJECT;
+    ASSERT_EQ("", ipTable.interpretPolicy());
+}
+
+
+TEST_F(IPTableEntryTest, TestIPTableEntryRules) {
+    IPTableEntry::Rule r;
+    r.host = "127.0.0.1/16";
+    r.ports.any = true;
+    r.ports.multiport = true;
+    r.ports.ports = "80,8080";
+    r.target = IPTableEntry::Target::ACCEPT;
+    //Test INPUT chain
+    //multiport list test
+    ASSERT_EQ("iptables -A INPUT -s 127.0.0.1/16 -p tcp --match multiport --sports 80,8080"
+            " -j ACCEPT", ipTable.interpretRule(r));
+
+    //multiport range test
+    r.ports.ports = "80:85";
+    ASSERT_EQ("iptables -A INPUT -s 127.0.0.1/16 -p tcp --match multiport --sports 80:85"
+            " -j ACCEPT", ipTable.interpretRule(r));
+
+    //sinlgeport test
+    r.ports.multiport = false; r.ports.ports = "80";
+    ASSERT_EQ("iptables -A INPUT -s 127.0.0.1/16 -p tcp --sport 80"
+            " -j ACCEPT", ipTable.interpretRule(r));
+
+    //Test OUTPUT chain
+    ipTable.m_type = "OUTPUT";
+    //multiport list test
+    r.ports.multiport = true; r.ports.ports = "80,8080";
+    ASSERT_EQ("iptables -A OUTPUT -d 127.0.0.1/16 -p tcp --match multiport --dports 80,8080"
+            " -j ACCEPT", ipTable.interpretRule(r));
+
+    //multiport range test
+    r.ports.ports = "80:85";
+    ASSERT_EQ("iptables -A OUTPUT -d 127.0.0.1/16 -p tcp --match multiport --dports 80:85"
+            " -j ACCEPT", ipTable.interpretRule(r));
+
+    //sinlgeport test
+    r.ports.multiport = false; r.ports.ports = "80";
+    ASSERT_EQ("iptables -A OUTPUT -d 127.0.0.1/16 -p tcp --dport 80"
+            " -j ACCEPT", ipTable.interpretRule(r));
+}
```

Comment:
Does these tests need to be run in a sequence? Otherwise split them into multiple tests.